src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,446 @@
+        builder.accept(MediaType.APPLICATION_JSON);
+        final String securityTokenValueStarter = "Token ";
+        builder.header(AUTHORIZATION_HEADER_NAME, securityTokenValueStarter + getAwsToken());
+        URI uri;
+        try {
+            final String yearUrlParam = "year=";
+            final String monthURLParam = "&month=";
+            uri = new URI(getAwsURL() + yearUrlParam + findBillingYear() + monthURLParam + findBillingMonthNumber());
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+        ClientRequest request = builder.build(uri, HTTP_METHOD_GET_NAME);
+        return request;
+    }
+    
+    public String findBillingYear() {
+        if (StringUtils.equalsIgnoreCase(getBillingPeriodParameterValue(), DEFAULT_BILLING_PERIOD_PARAMETER)) {
+            SimpleDateFormat dateFormater = new SimpleDateFormat("YYYY");
+            return dateFormater.format(new Date(Calendar.getInstance().getTimeInMillis()));
+        } else {
+            return StringUtils.split(getBillingPeriodParameterValue(), ",")[0];
+        }
+    }
+    
+    public String findBillingMonthNumber() {
+        if (StringUtils.equalsIgnoreCase(getBillingPeriodParameterValue(), DEFAULT_BILLING_PERIOD_PARAMETER)) {
+            Date currentDate = new Date(Calendar.getInstance().getTimeInMillis());
+            return String.valueOf(currentDate.getMonth()+1);
+        } else {
+            return StringUtils.split(getBillingPeriodParameterValue(), ",")[1];
+        }
+    }
+    
+    private String findBillingMonthName() {
+        Integer monthInt = new Integer(findBillingMonthNumber());
+        return new DateFormatSymbols().getMonths()[monthInt.intValue()-1];
+    }
+    
+    private List<AmazonAccountDetail> buildAmazonAcountListFromJson(String jsonResults) {
+        ObjectMapper mapper = new ObjectMapper();
+        AmazonAccountDetailContainer detailContainer = new AmazonAccountDetailContainer();
+        try {
+            detailContainer = mapper.readValue(jsonResults, AmazonAccountDetailContainer.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return detailContainer.getAccountDetail();
+    }
+
+    private boolean processAmazonAccountDetails(List<AmazonAccountDetail> amazonAccountDetails) {
+        boolean valid = true;
+        int count = 0;
+        for (AmazonAccountDetail amazonAccountDetail : amazonAccountDetails) {
+            if (StringUtils.equalsIgnoreCase(INTERNAL_KFS_ACCOUNT_DESCRIPTION, amazonAccountDetail.getKfsAccount())) {
+                LOG.info("processAmazonAccountDetails() Found an internal kfs account, ignoring this transaction for AWS account " 
+                    + amazonAccountDetail.getAwsAccount());
+            } else {
+                Account account = determineWhichAccountToUse(amazonAccountDetail.getCostCenter(), amazonAccountDetail.getKfsAccount());
+                if (account != null) {
```

Comment:
If the account is null, do we need to log that we skipped it or do anything else?